### PR TITLE
[Schemas] Connector SDK: Create and get a schema from Conduit

### PR DESCRIPTION
**Pull request by hariso**
### Description

Part of https://github.com/ConduitIO/conduit-connector-sdk/issues/132.

Changes in this PR:

* Adds a gRPC client for the schema service.
* Adds an in-memory schema service (used in testing)
* Renames schema's `name` field to `subject`.

Depends on https://github.com/ConduitIO/conduit-commons/pull/67.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-protocol/pulls) for the same update/change.
- [ ] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: go.mod
Comment (hariso) [inline go.mod:257]: Note: needs to be removed once the PR is approved.

**Review by raulb — APPROVED** [review]
@hariso LGTM

File: proto/conduit/v1/schema.proto
Comment (lovromazgon) [inline proto/conduit/v1/schema.proto:38]: At some point I heard you mention that we could rename `name` to `subject`, is this the plan here?

File: conduit/schema/client/client.go
Comment (lovromazgon) [inline conduit/schema/client/client.go:33]: As mentioned in the other review, I'd like to explore if there's a better way to expose the service rather than through the context. I'll try to not block this for too long.

File: proto/conduit/v1/schema.proto
Comment (hariso) [reply proto/conduit/v1/schema.proto:38]: I was thinking about that, but I'm leaning more towards keeping `name`. It's a bit more generic and given that we plan to move to our own schema registry.

Comment (lovromazgon) [reply proto/conduit/v1/schema.proto:38]: When we say "our own" it will still need to be compatible with the Confluent schema registry, otherwise we would be losing compatibility with other tools like Kafka Connect that only know how to talk to the Confluent schema registry.

I personally also prefer `name`, it's straight forward, don't get me wrong, but it might create unnecessary friction, since it's essentially the same as `subject` in the Confluent world.

Comment (hariso) [reply proto/conduit/v1/schema.proto:38]: I don't have much against `subject` and I don't think we're choosing between a good and a bad solution. But if we feel like something's better (the naming in this case), then I think we shouldn't be stuck with it because we think it might be confusing to users (especially because I think it will be an easy change).

Comment (lovromazgon) [reply proto/conduit/v1/schema.proto:38]: Hmm, actually, the more I think about it, the more I think `subject` is the better choice. I didn't care that much at first, because I thought it was an implementation detail (the friction I was thinking of was our own developer experience). But now I realized that this `name` is something the connector developer will see and use, while we will eventually expose the same field as `name` through the the schema registry API. It would be a bit confusing and inconsistent if we call the same field `name` in the connector SDK and `subject` in the schema registry API. Anyone with prior experience with the Confluent schema registry would also have an easier time if we used the same terminology across the board. It's not that it would be hard to get used to our terminology, but why create the barrier in the first place?

Comment (hariso) [reply proto/conduit/v1/schema.proto:38]: Personally, I don't think it's that much of a barrier, especially because there's no permanent context switching (from developing a connector to working with the schema registry directly). It's also pretty infrequently used.

I also think that that providing a slightly smoother switch to Conduit is less important that us having a more meaningful name long term.

But this way or the other, I don't think either provides so much difference that we should spend too much time on it, so AFAIC we can just migrate to `subject`. :+1:

Comment (lovromazgon) [reply proto/conduit/v1/schema.proto:38]: Thank you for indulging me 😄

Comment (hariso) [reply proto/conduit/v1/schema.proto:38]: lol :D I gotta return the favor of great reviews! :muscle:

File: proto/conduit/v1/schema.proto
Comment (lovromazgon) [inline proto/conduit/v1/schema.proto:16]: Can we move this `Schema` message to conduit-commons? We need it in processors as well.

Comment (raulb) [reply proto/conduit/v1/schema.proto:16]: @lovromazgon Added [here](https://github.com/ConduitIO/conduit-commons/pull/67/commits/20efa2b33ee80c327818950bafae9f6915db298b)

File: conduit/pschema/v1/fromproto/schema.go
Comment (lovromazgon) [inline conduit/pschema/v1/fromproto/schema.go:59]: The conversion from a proto schema to a Go schema (and back) should be moved to conduit-commons, so we have methods like `FromProto` and `ToProto` on `schema.Schema` (see how we do it on [`opencdc.Record`](https://github.com/ConduitIO/conduit-commons/blob/main/opencdc/proto.go)). This way the code can be reused in the processor SDK as well.

File: conduit/pschema/schema.go
Comment (lovromazgon) [inline conduit/pschema/schema.go:38]: We should reuse the [type](https://github.com/ConduitIO/conduit-commons/blob/0daeaaf8c952627b6f1ccd85c4904f02a75b1b9a/schema/schema.go#L19) we have in the commons package.

File: conduit/pschema/v1/grpc_client.go
Comment (lovromazgon) [inline conduit/pschema/v1/grpc_client.go:37]: We'll have to be able to set this through an argument.

File: conduit/pschema/v1/grpc_client.go
Comment (lovromazgon) [inline conduit/pschema/v1/grpc_client.go:50]: This will likely be in the error message already. That said, we should [unwrap the gRPC error](https://github.com/ConduitIO/conduit-connector-protocol/blob/c83ba94fe392faaaae56788f567b92692a10fe2e/pconnector/v2/client/errors.go#L37).

File: go.mod
Comment (lovromazgon) [inline go.mod:263]: This can be removed now.

File: conduit/pschema/in_memory.go
Comment (lovromazgon) [inline conduit/pschema/in_memory.go:60]: Let's check all the `name` occurrences (should be `subject`), I found this one, there might be more.

File: conduit/pschema/in_memory.go
Comment (lovromazgon) [inline conduit/pschema/in_memory.go:37]: Oh, seems like this should go to conduit-commons, we'll need it in the processor SDK (I assume we'll follow the same pattern as in the connector SDK).

File: conduit/pschema/v1/fromproto/schema.go
Comment (raulb) [reply conduit/pschema/v1/fromproto/schema.go]: @lovromazgon https://github.com/ConduitIO/conduit-commons/pull/72

File: pconduit/in_memory.go
Comment (lovromazgon) [inline pconduit/in_memory.go:33]: Hmm, I don't think the protocol is the right place for this. The purpose of this repo is to provide a thin layer around the protobuf files, while this part adds some actual logic. Granted, simple logic, but still, probably not the right place for this. Given that it will only be used by the connector SDK I'd move it there.

---

Sidenote: in the long run, my proposal would be to aim for something like this:
- implement a full-fledged schema service in a separate repo (e.g. `github.com/conduitio/schema-service`)
  - at first we'll probably do it directly in Conduit, but I think it would be worth extracting at some point, it can be a self contained piece of software
- the schema service should then provide different storage mechanisms, including an "in-memory" one
- we change this in-memory service to use the full-fledged schema service under the hood, so we're sure it works like the real one, because it is a real one

File: pconduit/doc.go
Comment (lovromazgon) [inline pconduit/doc.go:17]: This is not really true, also the package is `pconduit`. Do we even need this comment / file? 😬

File: internal/errors.go
Comment (lovromazgon) [inline internal/errors.go:37]: Now that I look at this, we shouldn't be reusing this in `pconduit`, as it can return an error coming from `pconnector`. Looks like we need separate utilities for `pconduit` and `pconnector` after all 😕

**Review by lovromazgon — COMMENTED** [review]
Good work @raulb, this already looks good. I'd still move some stuff around to make it _just right_ 👌

File: pconduit/doc.go
Comment (raulb) [reply pconduit/doc.go:17]: @lovromazgon i don't think we do.

File: internal/errors.go
Comment (raulb) [reply internal/errors.go:37]: @lovromazgon not sure I understand it. Wouldn't we unwrapping the error based on the request that errored (regardless of where this is actually coming from?)

**Pull request merged**
